erpnext/accounts/doctype/payment_entry/payment_entry.py

```diff
@@ -1698,13 +1698,42 @@ def get_outstanding_reference_documents(args, validate=False):
 	return data
 
 
-def split_invoices_based_on_payment_terms(outstanding_invoices, company):
-	invoice_ref_based_on_payment_terms = {}
+def split_invoices_based_on_payment_terms(outstanding_invoices, company) -> list:
+	"""Split a list of invoices based on their payment terms."""
+	exc_rates = get_currency_data(outstanding_invoices, company)
 
+	outstanding_invoices_after_split = []
+	for entry in outstanding_invoices:
+		if entry.voucher_type in ["Sales Invoice", "Purchase Invoice"]:
+			if payment_term_template := frappe.db.get_value(
+				entry.voucher_type, entry.voucher_no, "payment_terms_template"
+			):
+				split_rows = get_split_invoice_rows(entry, payment_term_template, exc_rates)
+				if not split_rows:
+					continue
+
+				frappe.msgprint(
+					_("Spliting {} {} into {} row(s) as per Payment Terms").format(
+						split_rows[0]["voucher_type"], split_rows[0]["voucher_no"], len(split_rows)
+					),
+					alert=True,
+				)
```

Comment:
Sry for nitpicking, but:

- Typo, it’s “Splitting”
- Using {0}, {1}, {2} is just a little bit more flexible in the long run.
- “Row(s)” is not necessary as it won’t ever be split into just 1 row…
- Don’t we have to translate the voucher_type as well?
- Maybe style the voucher_no in bold?
- Can’t we just use `entry.voucher_type` and `entry.voucher_no`? If yes: more simple. If not, sry.

Propose:

```
frappe.msgprint(
	_("Splitting {0} {1} into {2} rows as per Payment Terms").format(
		_(entry.voucher_type), frappe.bold(entry.voucher_no), len(split_rows)
	),
	alert=True,
)
```

P.S. In spite of minimal string changes, we may easily reuse all translations using `find` and `sed`. May be a followup though.

Reply:
Whatever, let‘s iterate over this in a followup.

Reply:
> “Row(s)” is not necessary as it won’t ever be split into just 1 row…

Grammatically no, technically yes. One of the payment terms (out of 2) could have outstanding 0, so only one payment term row is fetched. So technically the invoice was split but only one row was valid which contains a partial amount to the grand total. I think the no. of rows in this short-lived alert message is not very critical information right?

Changed it to:
<img width="436" alt="Screenshot 2023-11-09 at 12 48 24 PM" src="https://github.com/frappe/erpnext/assets/25857446/889dbe21-6034-4170-9382-0397c70486f4">

Reply:
> I think the no. of rows in this short-lived alert message is not very critical information right?

Short-lived, yes. But the moment you’re working with it, I think the No. of rows gives a better idea at what’s happened. Particularly if you have dozens of invoices in the table, this may be helpful, so I wouldn’t unnecessarily remove.

> One of the payment terms (out of 2) could have outstanding 0, so only one payment term row is fetched.

Wow, this seems wrong. At least it makes it really hard to understand what’s going on there and what happened to the remaining part of the invoice. I’d rather fetch all of them, no matter whether one may have outstanding 0.

But this should definitely be a followup. For now, I would really stick with the plural:

> Splitting {0} {1} into {2} rows as per Payment Terms

the singular being a rare edge case anyway, so the wording is optimal for the 99% regular cases.

Reply:
<img width="474" alt="Screenshot 2023-11-09 at 3 01 26 PM" src="https://github.com/frappe/erpnext/assets/25857446/4cf8bbe5-cdb7-4cbf-a280-23c505abca48">